### PR TITLE
Dynamo.Search Namespace refactor

### PR DESCRIPTION
**Pull request by mjkkirschner**
### Purpose:
http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-8862

to refactor Dynamo.Search Namespace, 

`Dynamo.Interfaces.ISource` is now `Dynamo.Search.ISource`
`Dynamo.Search.SearchElementGroup` is now `Dynamo.Search.SearchElements.SearchElementsGroup`
`Dynamo.Search.Interfaces.ISearchEntry` is now `Dynamo.Search.ISearchEntry`
`Dynamo.Search.Interfaces.ISearchCategory` is now `Dynamo.Search.ISearchCategory`

.cs files have been moved into folder structure that respects their namespaces

### Associated PRs
- DynamoRevit and Reach both still build with these changes. I do not think any associated PRs are necessary.

### Impact assessment

- [x] API change is added to the [API Changes document](https://github.com/DynamoDS/Dynamo/wiki/API-Changes).
- [ ] All tests pass using the self-service CI.

### Reviewers
@ikeough

### CODE REVIEW

File: src/DynamoCore/DynamoCore.csproj [hunk 1]
```diff
@@ -164,7 +164,7 @@ limitations under the License.
     <Compile Include="Core\AuthenticationManager.cs" />
     <Compile Include="Services\ShapewaysClient.cs" />
     <Compile Include="Models\NodeModelExtensions.cs" />
-    <Compile Include="Interfaces\ISource.cs" />
+    <Compile Include="Search\Interfaces\ISource.cs" />
```
Comment (ikeough) [inline src/DynamoCore/DynamoCore.csproj]: Let's get rid of the interfaces subfolder. I want namespaces that describe functionality, not type :)

File: src/DynamoCore/Search/Interfaces/ISource.cs [hunk 2]
```diff
@@ -1,6 +1,6 @@
 using System;
 
-namespace Dynamo.Interfaces
+namespace Dynamo.Search.Interfaces
```
Comment (ikeough) [inline src/DynamoCore/Search/Interfaces/ISource.cs]: See note above.

**Comment by ikeough** [comment]
@mjkkirschner One small comment, then LGTM.

**Comment by mjkkirschner** [comment]
@ikeough made the changes above, ptal

**Comment by ikeough** [comment]
@mjkkirschner  LGTM.

**Pull request merged**